src/components/Form/FormProvider.js

```diff
@@ -109,250 +109,278 @@ function getInitialValueByType(valueType) {
     }
 }
 
-function FormProvider({validate, formID, shouldValidateOnBlur, shouldValidateOnChange, children, formState, network, enabledWhenOffline, draftValues, onSubmit, ...rest}) {
-    const inputRefs = useRef({});
-    const touchedInputs = useRef({});
-    const [inputValues, setInputValues] = useState(() => ({...draftValues}));
-    const [errors, setErrors] = useState({});
-    const hasServerError = useMemo(() => Boolean(formState) && !_.isEmpty(formState.errors), [formState]);
+const FormProvider = forwardRef(
+    ({validate, formID, shouldValidateOnBlur, shouldValidateOnChange, children, formState, network, enabledWhenOffline, draftValues, onSubmit, ...rest}, forwardedRef) => {
+        const inputRefs = useRef({});
+        const touchedInputs = useRef({});
+        const [inputValues, setInputValues] = useState(() => ({...draftValues}));
+        const [errors, setErrors] = useState({});
+        const hasServerError = useMemo(() => Boolean(formState) && !_.isEmpty(formState.errors), [formState]);
 
-    const onValidate = useCallback(
-        (values, shouldClearServerError = true) => {
-            const trimmedStringValues = ValidationUtils.prepareValues(values);
+        const onValidate = useCallback(
+            (values, shouldClearServerError = true) => {
+                const trimmedStringValues = ValidationUtils.prepareValues(values);
 
-            if (shouldClearServerError) {
-                FormActions.setErrors(formID, null);
-            }
-            FormActions.setErrorFields(formID, null);
+                if (shouldClearServerError) {
+                    FormActions.setErrors(formID, null);
+                }
+                FormActions.setErrorFields(formID, null);
```

Comment:
Tiny issue - https://github.com/Expensify/App/issues/34684#issuecomment-1897457007

We should have re-rendered the form if the locale has changed.